test/e2e_node/memory_manager_test.go

```diff
@@ -0,0 +1,363 @@
+	framework.ExpectNoError(err, "failed to delete the state file")
+}
+
+type kubeletParams struct {
+	memoryManagerFeatureGate bool
+	memoryManagerPolicy      string
+	systemReservedMemory     []map[string]string
+	systemReserved           map[string]string
+	kubeReserved             map[string]string
+	evictionHard             map[string]string
+}
+
+func getUpdatedKubeletConfig(oldCfg *kubeletconfig.KubeletConfiguration, params *kubeletParams) *kubeletconfig.KubeletConfiguration {
+	newCfg := oldCfg.DeepCopy()
+
+	if newCfg.FeatureGates == nil {
+		newCfg.FeatureGates = map[string]bool{}
+	}
+	newCfg.FeatureGates["MemoryManager"] = params.memoryManagerFeatureGate
+	newCfg.MemoryManagerPolicy = params.memoryManagerPolicy
+
+	// update system-reserved
+	if newCfg.SystemReserved == nil {
+		newCfg.SystemReserved = map[string]string{}
+	}
+	for resourceName, value := range params.systemReserved {
+		newCfg.SystemReserved[resourceName] = value
+	}
+
+	// update kube-reserved
+	if newCfg.KubeReserved == nil {
+		newCfg.KubeReserved = map[string]string{}
+	}
+	for resourceName, value := range params.kubeReserved {
+		newCfg.KubeReserved[resourceName] = value
+	}
+
+	// update hard eviction threshold
+	if newCfg.EvictionHard == nil {
+		newCfg.EvictionHard = map[string]string{}
+	}
+	for resourceName, value := range params.evictionHard {
+		newCfg.EvictionHard[resourceName] = value
+	}
+
+	// update reserved memory
+	if newCfg.ReservedMemory == nil {
+		newCfg.ReservedMemory = []map[string]string{}
+	}
+	for _, p := range params.systemReservedMemory {
+		newCfg.ReservedMemory = append(newCfg.ReservedMemory, p)
+	}
+
+	return newCfg
+}
+
+func updateKubeletConfig(f *framework.Framework, cfg *kubeletconfig.KubeletConfiguration) {
+	// remove the state file
+	deleteMemoryManagerStateFile()
+
+	// Update the Kubelet configuration
+	framework.ExpectNoError(setKubeletConfiguration(f, cfg))
+
+	// Wait for the Kubelet to be ready.
+	gomega.Eventually(func() bool {
+		nodes, err := e2enode.TotalReady(f.ClientSet)
+		framework.ExpectNoError(err)
+		return nodes == 1
+	}, time.Minute, time.Second).Should(gomega.BeTrue())
+}
+
+func getAllNUMANodes() []int {
+	outData, err := exec.Command("/bin/sh", "-c", "lscpu").Output()
+	framework.ExpectNoError(err)
+
+	numaNodeRegex, err := regexp.Compile(`NUMA node(\d+) CPU\(s\):`)
+	framework.ExpectNoError(err)
+
+	matches := numaNodeRegex.FindAllSubmatch(outData, -1)
+
+	var numaNodes []int
+	for _, m := range matches {
+		n, err := strconv.Atoi(string(m[1]))
+		framework.ExpectNoError(err)
+
+		numaNodes = append(numaNodes, n)
+	}
+
+	sort.Ints(numaNodes)
+	return numaNodes
+}
+
+// Serial because the test updates kubelet configuration.
+var _ = SIGDescribe("Memory Manager [Serial] [Feature:MemoryManager][NodeAlphaFeature:MemoryManager]", func() {
+	// TODO: add more complex tests that will include interaction between CPUManager, MemoryManager and TopologyManager
```

Comment:
i would also add tests that do the following:

1) create more than one pod
2) create pods that dont fit
3) pods that use init containers

above should be done before moving out of alpha

Reply:
I will start to work on it.

Reply:
> 2\. create pods that dont fit

@derekwaynecarr  This one will require the multi NUMA host because when we are talking about a single NUMA node host, the pod will be rejected by the scheduler and not by the topology manager. I can add it, but it will be always skipped under the CI because k8s CI does not have support for multi-NUMA tests. I have some plans to add such support via fake NUMA kernel arguments, but it will need to introduce new functionality to the e2e node remote runner. Please give me know what do you prefer:
- to add it now, but it will be always skipped
- wait until we will have some infrastructure support to run multiple NUMA nodes tests

Reply:
add it now, and skip (prior to exit alpha).  downstream testing on alternative infrastructures would benefit.

Reply:
Already added it under - https://github.com/kubernetes/kubernetes/pull/95479/commits/da84cf6fea8faf1db61524c54ba9ff63bf2a2853#diff-a059dfe1f8609ac5352a3b897b3a64c682c8bac087aad0483e6bf8dd72a99850R459